src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs

```diff
@@ -55,6 +55,8 @@ internal ResourceReader(Stream stream, Dictionary<string, ResourceLocator> resCa
             "the user to only get one error.")]
         private object DeserializeObject(int typeIndex)
         {
+            Debug.Assert(Monitor.IsEntered(this));
```

Comment:
Can we also add a comment explaining why this is being added?

Reply:
After double checking I don't think we need this one; this method doesn't access `_store` directly. Removed